presto-main/src/main/java/com/facebook/presto/server/QuerySessionSupplier.java

```diff
@@ -161,12 +161,12 @@ public Session createSession(QueryId queryId, SessionContext context, Optional<S
                     queryType,
                     resourceGroupId);
             for (Entry<String, String> entry : sessionPropertyConfigurationManager.get().getSystemSessionProperties(configContext).entrySet()) {
-                sessionBuilder.setSystemProperty(entry.getKey(), entry.getValue());
+                sessionBuilder.setSystemProperty(entry.getKey().toLowerCase(Locale.US), entry.getValue().toLowerCase(Locale.US));
```

Comment:
Can you add some tests for this? Full integration tests (that run SQL) would be good, too.

Reply:
Why are we lowercasing the value? Only the keys are case insensitive. The value could be an arbitrary string (the semantics of which are up to the session property)

Reply:
You mean a test for toLowerCase()?

Reply:
Neither QueryAssertions nor AbstractTestQueries support SET SESSION, so I would have to build a bunch of infra.